extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -360,6 +379,51 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
     });
   }
 
+  /**
+   * Handles Fluid-related messages dispatched from SafeFrame.
+   * @param {!Event} event
+   */
+  receiveMessageForFluid(event) {
+    if (event.origin != 'https://tpc.googlesyndication.com') {
+      return;
+    }
+    const data = tryParseJson(getData(event));
+    if (!data || data['s'] != 'creative_geometry_update' ||
+        data['sentinel'] != this.sentinel) {
+      return;
+    }
+    // The first creative_geometry_update message will contain bad
+    // geometric data, as it will have been computed using the initial,
+    // incorrect, iframe style. We use this first message as a
+    // triggering point to restyle the iframe, which will in turn cause
+    // safeframe to send another creative_geometry_update message, this
+    // time containing the correct information.
+    const styleString = this.iframe.getAttribute('style');
+    if (/width: 0px/.test(styleString) &&
+        /height: 0px/.test(styleString)) {
+      setStyles(this.iframe, {
+        width: '100%',
+        height: '100%',
+        position: 'relative',
+      });
+      setStyles(this.element, {
+        width: computedStyle(this.win,
+            dev().assertElement(this.element.parentElement)).width,
+      });
+    } else {
+      const payload = tryParseJson(data['p']);
+      this.attemptChangeSize(payload['height'], undefined)
+          .then(() => {
+            if (this.fluidImpressionUrl_) {
+              this.fireDelayedImpressions(this.fluidImpressionUrl_, false);
+              this.fluidImpressionUrl_ = null;
+            }
+            this.onFluidResize();
+          })
+          .catch(() => this.forceCollapse());
```

Comment:
We decided we wouldn't try again later, right?  Perhaps TODO?  CSI ping?

Reply:
Added TODO.